### ISSUE TITLE
Batch inference is taking sequential inference time

### ISSUE THREAD

**Issue by arbitropy** [ultralytics/ultralytics#22383]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

I have a variable number of rtsp cameras i am reading frames from, and i want to do batch detection on all the frames at once. 
Sending them as a list takes sequential time, even though verbose log shows similar inference time for all the frames. But the actual time of the inference was >sequential processing. Here are some verbose predict logs and my own performance timing:
```
# 3 cameras at once
0: 736x1280 5 Cars, 13.0ms
1: 736x1280 5 Cars, 13.0ms
2: 736x1280 5 Persons, 10 Cars, 13.0ms
Speed: 3.5ms preprocess, 13.0ms inference, 0.9ms postprocess per image at shape (3, 3, 736, 1280)
2025-10-14 15:39:39.748 | INFO     | car-interaction:<module>:979 - Perf - FPS: 11.00 | Total: 90.91ms | Detect: 52.98ms 

# 7 cameras at once 
0: 736x1280 5 Cars, 13.4ms
1: 736x1280 6 Cars, 13.4ms
2: 736x1280 2 Persons, 9 Cars, 13.4ms
3: 736x1280 8 Cars, 13.4ms
4: 736x1280 2 Persons, 9 Cars, 13.4ms
5: 736x1280 3 Persons, 15 Cars, 13.4ms
6: 736x1280 8 Persons, 13.4ms
Speed: 4.6ms preprocess, 13.4ms inference, 0.7ms postprocess per image at shape (7, 3, 736, 1280)
2025-10-14 16:04:21.220 | INFO     | car-interaction:<module>:976 - Perf - FPS: 4.38 | Total: 228.55ms | Detect: 132.81ms
```

How do i enable true batch processing for production?
I have seen similar issue raised here #9838, but the solution provided there to send the frames as stacked in a single numpy array doesn't work. 

I have tried both inference using triton inference server and direct model loading, both has the same issue. 

### Environment

```
Ultralytics 8.3.168 🚀 Python-3.10.18 torch-2.8.0+cu128 CUDA:0 (NVIDIA GeForce RTX 5080, 15834MiB)
Setup complete ✅ (20 CPUs, 31.0 GB RAM, 241.3/914.8 GB disk)

OS                  Linux-6.14.0-33-generic-x86_64-with-glibc2.39
Environment         Linux
Python              3.10.18
Install             git
Path                /home/user/Projects/carshowroom_ai/car_interaction/.venv/lib/python3.10/site-packages/ultralytics
RAM                 30.97 GB
Disk                241.3/914.8 GB
CPU                 Intel Core(TM) Ultra 7 265KF
CPU count           20
GPU                 NVIDIA GeForce RTX 5080, 15834MiB
GPU count           1
CUDA                12.8

numpy               ✅ 2.2.6>=1.23.0
matplotlib          ✅ 3.10.7>=3.3.0
opencv-python       ✅ 4.11.0.86>=4.6.0
pillow              ✅ 11.3.0>=7.1.2
pyyaml              ✅ 6.0.3>=5.3.1
requests            ✅ 2.32.5>=2.23.0
scipy               ✅ 1.15.3>=1.4.1
torch               ✅ 2.8.0+cu128>=1.8.0
torch               ✅ 2.8.0+cu128!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.23.0+cu128>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 7.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.3.3>=1.1.4
ultralytics-thop    ✅ 2.0.17>=2.0.0
```

### Minimal Reproducible Example

```
from ultralytics import YOLOE
detector = YOLOE(model_path, task="segment")

all_frames = [list of frames to predict on]

detection_start_time = time.perf_counter()

batch_results = detector.predict(
    all_frames,
    verbose=True,
    half=True,
    classes=classes_to_detect,
    conf=detection_cfg.get('conf_threshold', 0.3),
    device=device,
    iou=detection_cfg.get('iou_threshold', 0.5),
    imgsz=detection_cfg.get('imgsz', 640),
    batch=len(all_frames),
)

detection_end_time = time.perf_counter()
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @arbitropy, thanks for reporting this and for using Ultralytics 🚀. This is an automated triage message to help move things forward quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

Based on your description (batch predict across multiple RTSP frames appearing sequential), could you please share the following to create a clear MRE?
- Exact minimal script that reproduces the timing issue (no external dependencies beyond ultralytics/torch), including how frames are prepared (list vs stacked array) and the exact `predict()` arguments you use
- Model path/checkpoint and task used, plus the number of input frames per call and their shapes
- A tiny set of synthetic or fixed sample frames (e.g., generated/random or a small subset) that reproduces the behavior
- Full verbose logs for a single run showing per-image and aggregate Speed lines
- Your timing code block and where it’s placed relative to model calls
- Whether the behavior also occurs in a clean environment using our notebook examples below

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` and ensure all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are installed in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/). This helps verify the behavior isn’t already resolved in the newest release:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

If possible, please attempt to reproduce the behavior in one of the environments above (Colab/Paperspace/Kaggle) using a minimal script. This helps us isolate environment-specific factors 🧪.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for the detailed environment info and logs you’ve already provided. Once we have an MRE that reproduces the timing gap you’re observing, we can investigate further and help determine next steps ✨

**Comment by Y-T-G** [comment]
What exact model are you using? Is this issue only with YOLOE? Are you using PyTorch model or did you export it to some other backend? What's the difference between total vs. detect time in your logs? What's your GPU utilization during batch inference?
